### ISSUE TITLE
Turning off all rules and then enabling desired rules

### ISSUE THREAD

**Issue by JamesNK** [dotnet/roslyn-analyzers#2380]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

Example: v2.9.2

#### Repro steps

1. Add package reference
1. Copy contents of `AllRulesDisabled.ruleset` to local file and reference from csproj
1. Compile

#### Expected behavior

No code analysis warnings because they're all disabled

#### Actual behavior

Code analysis warnings, e.g. warning CA2208: Call the ArgumentOutOfRangeException constructor that contains a message and/or paramName parameter.

**Comment by JamesNK** [comment]
I just want to disable all the rules and then choose the rules I want to opt-in to.

As far as I can tell Microsoft.CodeAnalysis.FxCopAnalyzers turns on everything by default.

**Comment by mavasani** [comment]
Yes, this is expected to work as you described. I’ll check whats broken here.

**Comment by mavasani** [comment]
Alright, I was able to get this to repro on the latest 16.1 preview bits, and this seems like a bug in our VS IDE project system where live analysis is always using the older snapshot of the ruleset file for intellisense diagnostics. A simpler repro is as below:

1. Create a C# class library project with the following source code:
```csharp
using System;
 
namespace Samples1
{    
    public class Book    
    {        
        private readonly string _Title;
 
        public Book(string title)        
        {            
            // Violates this rule (constructor arguments are switched)            
            if (title == null)                
                throw new ArgumentNullException("title cannot be a null reference (Nothing in Visual Basic)", "title");
 
            _Title = title;        
        }
 
        public string Title        
        {            
            get { return _Title; }        
        }
    }
}
```
2. Add a NuGet package reference to https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.2
3. Confirm CA2208 diagnostics in error list, both from build and intellisense.
4. From the solution explorer `Analyzers -> Microsoft.NetCore.Analyzers -> CA2208` node, set ruleset severity for CA2208 to `Error` and verify the diagnostics in error list are now errors.
5. Now open the ruleset created in Step 4 and manually added the CA2208 entry added at the end of the file to change it from `Error` to `None`.

**Expected**
No CA2208 diagnostics in error list from build or live analysis

**Actual**
CA2208 diagnostics are still shown in error list from live analysis. However, if you build the project and see the output window, there are no CA2208 diagnostics. Go back to the error list, still shows the old CA2208 diagnostics. Even if you edit the source code to rename the parameter name of the ctor from `title` to `title2`, the error list updates, but still shows newer CA2208 diagnostics. Build and switch error list to "Build Only" and verify no CA2208 diagnostics, but switching to "Intellisense Only" again shows CA2208 diagnostics.

This indicates that the project system is not using the latest contents of the ruleset for live analysis, leading to this bad user experience. Tagging @jasonmalinowski @jinujoseph - can we port this issue to Roslyn and try to get it fixed for 16.1 preview2? I agree with @JamesNK that the experience is pretty terrible in the current state.

**Comment by mavasani** [comment]
Btw, James, you can workaround this bug by closing and reopening the solution.
